MParT/TrainMap.h

```diff
@@ -0,0 +1,40 @@
+#ifndef MPART_TRAINMAP_H
+#define MPART_TRAINMAP_H
+
+#include <functional>
+#include <nlopt.hpp>
+#include "ConditionalMapBase.h"
+#include "MapObjective.h"
+
+namespace mpart {
+struct TrainOptions {
+    std::string opt_alg = "LD_LBFGS";
+    double opt_stopval = -std::numeric_limits<double>::infinity();
+    double opt_ftol_rel = 1e-3;
+    double opt_ftol_abs = 1e-3;
+    double opt_xtol_rel = 1e-4;
+    double opt_xtol_abs = 1e-4;
+    int opt_maxeval = 30;
```

Comment:
30 seems really small for a maxeval....  How about 1000?

Reply:
I've been finding that 30 is usually plenty for my scenarios (which are admittedly a little low on dimension), but we can change it if you want. If it's any consolation, this returns an improved map after it finishes, so if you forget to increase this, you can just throw it back into `TrainMap` a second time.